unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SettingsPanelHUD/Scripts/ControlsModule/SettingsControlView.cs

```diff
@@ -16,7 +17,11 @@ public interface ISettingsControlView
         /// </summary>
         /// <param name="controlConfig">Model that will contain the configuration of the CONTROL.</param>
         /// <param name="settingsControlController">Controller associated to the CONTROL view.</param>
-        void Initialize(SettingsControlModel controlConfig, SettingsControlController settingsControlController);
+        void Initialize(
```

Comment:
I agree with @pbosio comment. The settings apply should be completely decoupled from the view. But, I think that moving the settings application back to `GeneralSettingsController`/`QualitySettingsController` is not the proper way to go towards this direction. The problem with this is that we are still centralizing the settings code into those classes, and the maintenance for adding a new setting still depends on wiring those two classes into the controllers. These controllers will keep growing over time and the application domain will not match the design, as we can chose to split our settings into more categories than "General" and "Quality".

To improve this design, the setting controllers should be completely detached from the view existence so they can be applied without instancing the view.

After looking at the code, I can propose the following:

- Removing the `view`, `generalSettingsController` and `qualitySettingsController` references from each `SettingsControlController`.
- The sole responsibility of `SettingsControlController` should be to apply the setting and storing the current value for saving it later.
- controller-view relationship with `SettingsControlController` should be inverted using events, so it can be completely abstracted from the view. The views should subscribe to controller events to change state.
- Then, we can have a `SettingsControlController` collection on a composite class, so we can serialize the controller to save/load the data. Maybe we can do this on the `Settings` class.

After this is done, we should be able to just remove `IGeneralSettingsController` and `IQualitySettingsController` and at the same time keep the controllers abstracted from the views. With this approach, the settings would be modularized by the single responsibility principle. We could add new settings in a simpler way.

Just came with this proposal after looking quickly into the code, if the proposal doesn't match 100% I'm sure we can find another similar way to come up with an alternative that can let us go into the more modularized direction. 

What do you think?